### PR TITLE
support warnings for extra attrs and axis in non default value

### PR DESCRIPTION
**Pull request by veyron95**
### PR types

Others
### PR changes

Others
### Describe

Pcard-66964
support warnings for extra attrs and axis in non default value

### CODE REVIEW

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -3064,6 +3080,11 @@ def find_name(var_list, name):
                     attr_val = op_attrs[attr_name]
                     self._update_desc_attr(attr_name, attr_val)
                 for attr_name in extra_attrs_map.keys():
+                    if _enable_printing_extra_attrs_:
+                        warnings.warn(
+                            "op %s has extra_attr: %s" % (type, attr_name)
```
Comment (JiabinYang) [inline python/paddle/fluid/framework.py]: use..

**Review by JiabinYang — COMMENTED** [review]
one comments

File: python/paddle/fluid/framework.py [hunk 2]
```diff
@@ -127,6 +127,54 @@ def __setattr__(self, name, val):
     'FLAGS_CUDA_GRAPH_USE_STANDALONE_EXECUTOR', 0
 )
 
+
+_enable_printing_extra_attrs_ = os.environ.get('FLAGS_print_extra_attrs', 0)
+
+if _enable_printing_extra_attrs_:
+    special_ap_attrs = {
```
Comment (JiabinYang) [inline python/paddle/fluid/framework.py]: op

Comment (veyron95) [reply python/paddle/fluid/framework.py]: done

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (veyron95) [reply python/paddle/fluid/framework.py]: done

**Review by JiabinYang — DISMISSED** [review]
LGTM

File: python/paddle/fluid/framework.py [hunk 3]
```diff
@@ -127,6 +126,53 @@ def __setattr__(self, name, val):
...
+    "elementwise_sub": [{"axis": -1}],
+    "elementwise_mul": [{"axis": -1}],
+    "elementwise_div": [{"axis": -1}],
+    "elementwise_min": [{"axis": -1}],
+    "elementwise_pow": [{"axis": -1}],
+    "elementwise_div": [{"axis": -1}],
+    "elementwise_mod": [{"axis": -1}],
+    "elementwise_floordiv": [{"axis": -1}],
+    "elementwise_max": [{"axis": -1}],
+    "elementwise_min": [{"axis": -1}],
```
Comment (zyfncg) [inline python/paddle/fluid/framework.py]: elementwise_min 重复了

File: python/paddle/fluid/framework.py [hunk 4]
```diff
@@ -127,6 +126,53 @@ def __setattr__(self, name, val):
...
+    "equal": [{"axis": -1}],
+    "not_equal": [{"axis": -1}],
+    "amax": [{"reduce_all": False}],
+    "amin": [{"reduce_all": False}],
+    "any": [{"reduce_all": False}],
+    "amax": [{"reduce_all": False}],
+    "frobenius_norm": [{"reduce_all": False}],
+    "logsumexp": [{"reduce_all": False}],
+    "reduce_max": [{"reduce_all": False}],
+    "reduce_max": [{"reduce_all": False}],
```
Comment (zyfncg) [inline python/paddle/fluid/framework.py]: reduce_max -> reduce_min

Comment (veyron95) [reply python/paddle/fluid/framework.py]: done, thx

File: python/paddle/fluid/framework.py (hunk 3 above)
Comment (veyron95) [reply python/paddle/fluid/framework.py]: done, thx

**Review by zyfncg — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Pull request merged**